packages/flutter_tools/lib/src/commands/build_ipa.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../application_package.dart';
+import '../base/common.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import '../base/process.dart';
+import '../build_info.dart';
+import '../convert.dart';
+import '../globals.dart' as globals;
+import '../ios/xcodeproj.dart';
+import '../runner/flutter_command.dart' show FlutterCommandResult;
+import 'build.dart';
+
+/// Builds an iOS .ipa package for App Store submission.
+///
+/// A wrapper around "xcodebuild -exportArchive". Can only be run on a macOS host.
+class BuildIPACommand extends BuildSubCommand {
+  BuildIPACommand({
+    @visibleForTesting Platform platform,
+    @visibleForTesting FileSystem fileSystem,
+    @visibleForTesting Logger logger,
+    @visibleForTesting ProcessManager processManager,
+    @visibleForTesting XcodeProjectInterpreter xcodeProjectInterpreter,
+  })  : _platform = platform,
+        _xcodeProjectInterpreter = xcodeProjectInterpreter,
+        _fileSystem = fileSystem,
+        _logger = logger,
+        _processUtils =
+            ProcessUtils(processManager: processManager, logger: logger) {
+    argParser.addOption(
+      'export-options-plist',
+      valueHelp: 'ExportOptions.plist',
+      help:
+          'Path to the "-exportOptionsPlist" file passed to Xcode. See "man xcodebuild" for available keys.',
+    );
+    // Flavors may have different target names, and therefore different archive output paths.
+    usesFlavorOption();
+  }
+
+  Platform _platform;
+  XcodeProjectInterpreter _xcodeProjectInterpreter;
+  FileSystem _fileSystem;
+  Logger _logger;
+  ProcessUtils _processUtils;
+
+  @override
+  final String name = 'ipa';
+
+  @override
+  final String description =
+      'Build an iOS App Store package after running "flutter build xcarchive" (Mac OS X host only).';
```

Comment:
Makes sense that it's 2 separate steps in xcode and it's safer to model our modularity after the underlaying tool. But on the frontend, we don't have to expose the same granularity to the users I suspect. 

Could we just wrap the inner workings?

Reply:
It sounds like that's what you and @jonahwilliams would prefer.

I could convert this command into something that runs in response to the export plist being passed into the `xcarchive` command?  So no `flutter build ipa` but instead `flutter build xcarchive --exportOptionsPlist <file>` that creates the xcarchive and then makes an ipa out of it?

Reply:
Done.